[CI] MixedClusterClientYamlTestSuiteIT test {p0=index/100_field_name_length_limit/Test field name length limit array} failing

**Build Scans:**
- [elasticsearch-intake #38202 / 9.3.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/e3emtk6uu463i)
- [elasticsearch-intake #38166 / 9.4.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/nc4bor3qlucki)
- [elasticsearch-pull-request #139169 / 9.4.0_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/qmirafcs3ui2e)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.3.4#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=index/100_field_name_length_limit/Test field name length limit array}" -Dtests.seed=29DF5408BF0E5937 -Dtests.bwc=true -Dtests.locale=mg-MG -Dtests.timezone=Pacific/Noumea -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dindex%2F100_field_name_length_limit%2FTest%20field%20name%20length%20limit%20array%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: field [test_index.mappings.properties.foo.properties.field_name_not_ignored.type] is null
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=index/100_field_name_length_limit/Test field name length limit array}
- [main] 1 consecutive failures in step 9.3.4_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test test {p0=index/100_field_name_length_limit/Test field name length limit array} (0.5% fail rate in 571 executions)
- [main] 2 failures in pipeline elasticsearch-intake (2.3% fail rate in 88 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -253,9 +253,6 @@ tests:
   method: testMultipleEntriesForSameMetricAcrossStreams
   issue: https://github.com/elastic/elasticsearch/issues/147170
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=index/100_field_name_length_limit/Test field name length limit array}
-  issue: https://github.com/elastic/elasticsearch/issues/147205
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only}
   issue: https://github.com/elastic/elasticsearch/issues/147251
 - class: org.elasticsearch.xpack.logsdb.RandomizedRollingUpgradeIT
@@ -273,9 +270,6 @@ tests:
 - class: org.elasticsearch.index.reindex.ReindexRelocationOnShutdownIT
   method: testReindexTaskRelocatesOnNodeShutdown
   issue: https://github.com/elastic/elasticsearch/issues/147449
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=index/100_field_name_length_limit/Test field name length limit}
-  issue: https://github.com/elastic/elasticsearch/issues/147494
 - class: org.elasticsearch.xpack.core.transform.transforms.pivot.ScriptConfigTests
   method: testFromXContent
   issue: https://github.com/elastic/elasticsearch/issues/147543

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/100_field_name_length_limit.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/100_field_name_length_limit.yml
@@ -37,6 +37,10 @@ setup:
   - match: { result: created }
 
   - do:
+      cluster.health:
+        wait_for_events: languid
+
+  - do:
       indices.get_mapping:
         index: test_index
 
@@ -88,6 +92,10 @@ setup:
             field_name_not_ignored: [ 10, 20, 30 ]
 
   - match: { result: created }
+
+  - do:
+      cluster.health:
+        wait_for_events: languid
 
   - do:
       indices.get_mapping:

